### ISSUE TITLE
android - locally generated files in favourite folders not synced

### ISSUE THREAD

**Issue by cast-away** [nextcloud/android#2395]
### Actual behaviour
- files (and folders) locally copied into a favourite folder by apps other than the nextcloud client are not detected by the sync client and thus not uploaded onto the server 

### Expected behaviour
- locally copied files onto favourite folders should be synced
 
### Steps to reproduce
1. Mark a folder as favorite through the Android client e.g. files/test/
2. Locate the folder in the phone local storage, e.g. storage/emulated/0/Android/media/com.nextcloud.client/nextcloud/user@server/files/test/
3. Copy a new file (e.g. newfile.txt) via another app (e.g. file browser) into the folder in point 2
4. With the nextcloud Android client, sync the folder previously marked as favourite. newfile.txt does not show up. 

### Environment data
Android version:  7.0

Device model: Motorola G4 Plus

Stock or customized system: stock

Nextcloud app version: 3.03

Nextcloud server version: 13

**Comment by ardevd** [comment]
This behaviour is not supported by the server application so I'd assume its not supported by the Android app either though perhaps @mario or @tobiasKaminsky can confirm?

**Comment by mario** [comment]
This has nothing to do with the server. We currently do NOT sync favorite stuff at all, but instead, you mark the file as "Keep available offline" and then it's synced. It does not support the folder mechanism, and it is indeed our goal to move to favorites-based syncing, including folders support.

**Comment by AndyScherzinger** [comment]
@mario @ardevd I think the described request here is different and will imho never be supported since in my personal opinion this is against Android principal.

From what I understood the request is:
Locally copying files within the Nextcloud app's installation/data folder should be synced to the server, like the desktop client does. While this is fine for desktop where the client is a file browser Integration the Android client is an app with a protected data folder and nobody except the app itself should write to that storage.

Just my two cents.

This might get better when we get to the point of having the time to implement a write document-provider.

**Comment by ardevd** [comment]
Yep, that's my understanding as well. It doesn't really make sense to support direct file system manipulations in my opinion, though I do see how some users would prefer to have that ability.

**Comment by cast-away** [comment]
I think I see your point, but isn't this exactly what the folders in auto upload do? In that instance it is not just the android app being able to write there as the pictures are generated by another app?
I might be misunderstanding how this should work though...

**Comment by AndyScherzinger** [comment]
@cast-away yes and no... The auto upload, especially when expert mode is turned on can watch on any folder on the device and upload new/changed files. So if you need that kind of feature it is best to use the auto upload in expert mode with custom folders (for any kind of file type). As for the storage path of the app itself this shouldn't be done simply since Android considers this a protected space where also the write permissions get less and less with each Android OS release. 
Having said this it is also not recommended to set up auto upload for a data folder of another app while this might work for a certain time it might fail in the future with any new Android OS release.

In short, if you need a folder to be watched on your device, activate the expert mode in the Nextcloud app's settings, go back to auto upload and setup a so called custom folder where you define local and remote folders and when this should get uploaded (wifi-only, charging-only etc.)

**Comment by AndyScherzinger** [comment]
@mario @tobiasKaminsky what's your vote on this? I vote close, since I don't think we'll should implement this since imho it is against Android OS principles.

**Comment by cast-away** [comment]
Thanks Andy, I did not know expert mode unlocked the option to set up a custom folder pair. Will give it a go.
